### PR TITLE
Add acceptance tests for teams

### PR DESCRIPTION
**Pull request by chloeruka**
This is one of the resources that does not yet have acceptance tests associated with it. In order to guard against regressions, I have added the following tests:

* creating a team (with only the required attributes)
* updating a team (changing its name, description and visibility)
* importing a team

During this work I discovered an error in importing teams. In order to resolve this, I added the DefaultMemberRole value which was present in the schema to the TeamNode struct, and added a mapping to updateTeam, so this PR doubles as a bugfix for importing teams.

### CODE REVIEW

File: buildkite/resource_team_test.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+			return fmt.Errorf("Error fetching team from graphql API: %v", err)
+		}
+
+		if string(query.Node.Team.ID) == "" {
+			return fmt.Errorf("No team found with graphql id: %s", resourceState.Primary.ID)
+		}
+
+		if string(query.Node.Team.Name) != resourceState.Primary.Attributes["name"] {
+			return fmt.Errorf("Team name in state doesn't match remote name")
+		}
```
Comment (yob) [inline buildkite/resource_team_test.go:145]: In our other resource tests I think we only checked computer properties in this function, and I also note that name is checked in `testAccCheckTeamRemoteValues`. Do we definitely want to check it here too?

**Comment by yob** [comment]
> During this work I discovered an error in importing teams. In order to resolve this, I added the DefaultMemberRole value which was present in the schema to the TeamNode struct, and added a mapping to updateTeam, so this PR doubles as a bugfix for importing teams.

Nice, tests work!

Purely out of interesting, could you describe this bug in a bit more detail? Did it manifest as an error when updating a team during `terraform apply`?

File: buildkite/resource_team_test.go (hunk 1 above)
Comment (chloeruka) [reply buildkite/resource_team_test.go:145]: Probably excessive, I'll remove it!

**Comment by chloeruka** [comment]
> Purely out of interesting, could you describe this bug in a bit more detail? Did it manifest as an error when updating a team during `terraform apply`?

image

When importing a resource, it was failing on `ImportStateVerify: true` because the returned resource did not have the field `DefaultMemberState` present in the definition of the resource:

```HCL
resource "buildkite_team" "test" {
  name = "example"
  description = "a cool team of example"
  privacy = "VISIBLE"
  default_team = true
  default_member_role = "MEMBER" # this field
}
```

**Comment by yob** [comment]
Oh, right. So the schema had `default_member_role` and the provider would include it in team updates/creates, but it would never read the value back from buildkite and update the state?

Cool, nice find :sparkles:

**Review by yob — APPROVED** [review]
ship it :+1:

**Pull request merged**